blog/_posts/2018-03-29-missing.md

```diff
@@ -0,0 +1,394 @@
+
+    julia> x = 1:3
+    1:3
+
+    julia> x[[true, missing, false]]
+    ERROR: ArgumentError: unable to check bounds for indices of type Missing
+
+    julia> x[[1, missing]]
+    ERROR: ArgumentError: unable to check bounds for indices of type Missing
+
+Convenience functions are provided to get rid of missing values explicitly.
+First, the `skipmissing` function returns an iterator over the non-missing values
+in the passed collection. It is particularly useful to ignore missing values when
+computing reductions. Call `collect` to obtain a vector with all non-missing values.
+
+    julia> sum(skipmissing([1, missing, 2]))
+    3
+
+    julia> mean(skipmissing([1, missing, 2]))
+    1.5
+
+    julia> collect(skipmissing([1, missing, 2]))
+    2-element Array{Int64,1}:
+    1
+    2
+
+Second, the `coalesce` function returns the first non-missing argument, which
+as a special case allows replacing missing values with a particular value.
+Combined with the "dot" broadcasting syntax, it allows replacing all missing
+values in an array:
+
+    julia> coalesce(missing, 0)
+    0
+
+    julia> coalesce(missing, missing, 0)
+    0
+
+    julia> coalesce.([1, missing, 2], 0)
+    3-element Array{Int64,1}:
+    1
+    0
+    2
+
+    julia> coalesce.([1, missing, 2], [2, 3, missing])
+    3-element Array{Int64,1}:
+    1
+    3
+    2
+
+A restricted set of functions and operators follow different semantics than
+those described above. They can be grouped into three classes:
+
+- `ismissing` returns `true` if the input is `missing`, and `false` otherwise.
+
+- `===`, `isequal` and `isless` always return a Boolean. `===` and `isequal` return
+  `true` when comparing `missing` to `missing`, and `false` otherwise.
+  `isless` also belongs to this class, and returns `true` when
+  comparing any non-missing value to `missing`, and `false` otherwise: missing values
+  are sorted last.
+
+- `&`, `|` and `⊻`/`xor` implement [three-valued logic](https://en.wikipedia.org/wiki/Three-valued_logic),
+  returning either a Boolean value or `missing` depending on whether the result is
+  fully determined even without knowing what could be the value behind `missing`.
+
+- `==`, `<`, `>`, `<=` and `>=` return `missing` if one of the operands is `missing`
+  just like any other operators. When called on collections containing missing values,
+  these operators are applied recursively and also follow three-valued logic: they
+  return `missing` if the result would depend on what value a `missing` element would take.
+  This also applies to `all`, `any` and `in`.
+
+Short-circuiting operators `&&` and `||`, just like `if` conditions, throw an error
+if they need to evaluate a missing value.
+
+See the [manual](https://docs.julialang.org/en/latest/manual/missing/) for more details
+and illustrations about these rules. Let us note that they follow very closely those
+implemented by SQL's `NULL` and R's `NA`, making it easy to translate Julia code into
+SQL requests.
```

Comment:
This seems like a really important design decision and should be mentioned at the very top in my opinion

Reply:
Yeah, I agree. I've added another mention in the intro.